### PR TITLE
[dev-infra][build] Create a wrapper script for aws codebuild

### PR DESCRIPTION
**Pull request by ankushagarwal**
## Summary

build-aws.sh is a wrapper script which submits a build job to aws codebuild and tracks its progress. To submit a build, one needs to provide either the PR number or any valid git ref (commit id, branch name, etc)

## Test Plan

Ran locally

### CODE REVIEW

File: docker/validator-dynamic/build-aws.sh [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+  echo "Usage: build-aws.sh --version pull/123"
+  exit 1
+}
+
+while [[ "$1" =~ ^- ]]; do case $1 in
+  --version )
+    shift;
+    if [[ "$1" =~ ^pull ]]; then
+      SOURCE_VERSION="refs/${1}/head"
+      TAGS="dev_pr_${1/\//_}"
```
Comment (sausagee) [inline docker/validator-dynamic/build-aws.sh]: Something to consider: if the PR has updates, the prior image would become untagged. ECR would look littered.

File: docker/validator-dynamic/build-aws.sh [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+# build-aws.sh --version pull/123 # Builds the image from pull request 123 and tags it with dev_pr_123
+# build-aws.sh --version u29a020 # Builds the image from commit u29a020 and tags it with dev_u29a020
+
+if ! which jq &>/dev/null; then
+  echo "jq is not installed. Please install jq"
+  exit 1
+fi
+
+function print_usage() {
+  echo "Usage: build-aws.sh --version pull/123"
```
Comment (sausagee) [inline docker/validator-dynamic/build-aws.sh]: Add example for commit hash?

File: docker/validator-dynamic/build-aws.sh (hunk 1 above)
Comment (andll) [reply docker/validator-dynamic/build-aws.sh]: Should we do smth like `$User_$ShortGitHash` instead as a tag?

File: docker/validator-dynamic/build-aws.sh (hunk 2 above)
Comment (ankushagarwal) [reply docker/validator-dynamic/build-aws.sh]: Added.

File: docker/validator-dynamic/build-aws.sh (hunk 1 above)
Comment (ankushagarwal) [reply docker/validator-dynamic/build-aws.sh]: I have updated it such that every build also gets tagged with `dev_${ShortGitHash}`. This should be always unique and with this we don't need the `$User` prefix.  This should also avoid the ECR littering issue.

I have also added a way to add custom tags with `--tags` which can be used to add arbitrary user tags to every build.

**Pull request closed without merging**